ThumbnailSequence.cpp

```diff
@@ -1560,7 +1560,7 @@ ThumbnailSequence::CompositeItem::CompositeItem(
 bool
 ThumbnailSequence::CompositeItem::incompleteThumbnail() const
 {
-	return dynamic_cast<IncompleteThumbnail*>(m_pThumb) != 0;
+	return m_pThumb == 0 && dynamic_cast<IncompleteThumbnail*>(m_pThumb) != 0;
```

Comment:
You must be using std::move() incorrectly. In the constructor you should be doing something like:
: m_ptrDbg(std::move(dbg))
Where both m_ptrDbg and dbg are instances of std::unique_ptr().

Speaking of -Wl,-export-dynamic, you actually need to rebuild Qt with that as well. FreeBSD developers decided it's OK to break the standard behaviour for the sake of performance, as long as you can fix it with some compiler flags. I don't think this will necessarily help though, as the bug in question talks about dynamic_cast<> producing incorrect results rather than crashing.